Adding open Cypher docs/pages to GSQL Docs (#118)
tigergraph/gsql-docs: modules/appendix/pages/openCypher-gsql-from-clause-extension.adoc

= Cypher Pattern Support in GSQL

> We have extended GSQL to support Cypher patterns in the FROM clause.

Reviewer: ```suggestion
We have extended GSQL to support openCypher patterns in the FROM clause.
```

Author: Changed several places where I said "Cypher" to "openCypher". reflected in the latest commit.

GSQL maintains its SELECT-FROM-WHERE block structure, but TigerGraph now allows the following new syntax in the FROM clause

NOTE: This is in addition to the standard GSQL patterns, which will continue to be supported

== Cypher's vertex pattern notation

`(v:VType)` which now gets translated internally to standard GSQL's  `VType:v`.

Both `v` and `VType` are optional.

`VType` can be a disjunction.

== Cypher's edge patterns notation

`[e:EType]` which now gets translated internally to standard GSQL's `(EType:e)`.

Both `e` and `EType` are optional.

`EType` can be a disjunction

== Cypher's property condition notation

> This is applicable to both vertex and edge patterns. It has a general form:
> 
>     x {prop_1: val_1, …, prop_n: val_n}
> 
> and it now gets translated internally into a `WHERE` clause condition:

Reviewer: ```suggestion
GSQL handles openCypher's property condition condition

    x {prop_1: val_1, …, prop_n: val_n}

by translating into a GSQL `WHERE` clause condition:
```

Author: Reworded this in the latest commit.
